src/servers/src/auth/user_provider.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright 2022 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io;
+use std::io::BufRead;
+use std::path::Path;
+
+use async_trait::async_trait;
+use digest;
+use digest::Digest;
+use sha1::Sha1;
+use snafu::{OptionExt, ResultExt};
+
+use crate::auth::{
+    Error, IOErrSnafu, Identity, InvalidConfigSnafu, Password, UserInfo, UserNotFoundSnafu,
+    UserPasswordMismatchSnafu, UserProvider,
+};
+
+impl TryFrom<&str> for MemUserProvider {
+    type Error = Error;
+
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        let (mode, content) = value.split_once(':').context(InvalidConfigSnafu {
+            value: value.to_string(),
+            msg: "MemUserProviderOption must be in format `<option>:<value>`",
+        })?;
+        return match mode {
```

Comment:
`mode` needs "to_lowercase"?

Reply:
I don't think calling `to_lowercase` is a good idea, it should be case-sensitive, and the user must take care of it.